tests/testthat/test-read_cargo_metadata.R

```diff
@@ -0,0 +1,14 @@
+test_that("read_cargo_metadata() returns crate or workspace metadata", {
+  skip_if_not_installed("usethis")
+
+  path <- local_package("testpkg")
+
+  # capture setup messages
+  withr::local_options(usethis.quiet = FALSE)
+
+  use_extendr(path, quiet = TRUE)
+
+  out <- read_cargo_metadata(path)
+
+  expect_type(out, "list")
```

Comment:
Can we have a more meaningful test here please?

Reply:
Instead of using a snapshot, I extracted some metadata where we know what the value will be and did `expect_equal()`. If you would still like me to do a snapshot, I'm happy to do it!